File: benchmarks/zig/fibonacci.zig
// Fibonacci benchmark - tests recursive function calls
const std = @import("std");

fn fib(n: i64) i64 {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    const args = try std.process.argsAlloc(std.heap.page_allocator);
    defer std.process.argsFree(std.heap.page_allocator, args);

    var n: i64 = 35;
    if (args.len > 1) {
        n = try std.fmt.parseInt(i64, args[1], 10);
    }

    var timer = try std.time.Timer.start();
    const result = fib(n);
    const elapsed = timer.read() / std.time.ns_per_ms;

    try stdout.print("fib({d}) = {d}\n", .{ n, result });
    try stdout.print("Time: {d} ms\n", .{elapsed});
}

File: benchmarks/java/Fibonacci.java
// Fibonacci benchmark - tests recursive function calls
public class Fibonacci {
    public static long fib(int n) {
        if (n <= 1) return n;
        return fib(n - 1) + fib(n - 2);
    }

    public static void main(String[] args) {
        int n = 35;
        if (args.length > 0) n = Integer.parseInt(args[0]);

        long start = System.nanoTime();
        long result = fib(n);
        long elapsed = (System.nanoTime() - start) / 1000000;

        System.out.println("fib(" + n + ") = " + result);
        System.out.println("Time: " + elapsed + " ms");
    }
}
